File: apps/chat/libs/chat-storage/src/main/java/com/example/chat/storage/entity/ChatFriendshipEntity.java
package com.example.chat.storage.entity;

import java.time.Instant;

import com.example.chat.domain.friendship.FriendshipStatus;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.EnumType;
import jakarta.persistence.Enumerated;
import jakarta.persistence.Id;
import jakarta.persistence.Index;
import jakarta.persistence.PrePersist;
import jakarta.persistence.PreUpdate;
import jakarta.persistence.Table;
import jakarta.persistence.UniqueConstraint;
import lombok.AccessLevel;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

/**
 * 친구 관계 JPA Entity
 */
@Entity
@Table(name = "chat_friendships", indexes = {
        @Index(name = "idx_user_id", columnList = "user_id"),
        @Index(name = "idx_friend_id", columnList = "friend_id"),
        @Index(name = "idx_user_status", columnList = "user_id, status")
}, uniqueConstraints = {
        @UniqueConstraint(name = "uk_friendship", columnNames = { "user_id", "friend_id" })
})
@Getter
@Setter
@NoArgsConstructor(access = AccessLevel.PROTECTED)
@AllArgsConstructor(access = AccessLevel.PRIVATE)
@Builder
public class ChatFriendshipEntity {
    @Id
    @Column(name = "id", length = 36, nullable = false)
    private String id;

    @Column(name = "user_id", length = 36, nullable = false)
    private String userId;

    @Column(name = "friend_id", length = 36, nullable = false)
    private String friendId;

    @Enumerated(EnumType.STRING)
    @Column(name = "status", length = 20, nullable = false)
    private FriendshipStatus status;

    @Column(name = "nickname", length = 100)
    private String nickname;

    @Column(name = "favorite", nullable = false)
    @Builder.Default
    private boolean favorite = false;

    @Column(name = "created_at", nullable = false)
    private Instant createdAt;

    @Column(name = "updated_at", nullable = false)
    private Instant updatedAt;

    @PrePersist
    public void prePersist() {
        Instant now = Instant.now();
        if (createdAt == null)
            createdAt = now;
        if (updatedAt == null)
            updatedAt = now;
    }

    @PreUpdate
    public void preUpdate() {
        updatedAt = Instant.now();
    }
}

File: apps/chat/libs/chat-storage/src/main/kotlin/com/example/chat/storage/entity/ChatFriendshipEntity.kt
package com.example.chat.storage.entity

import com.example.chat.domain.friendship.FriendshipStatus
import jakarta.persistence.*
import java.time.Instant

/**
 * 친구 관계 JPA Entity
 */
@Entity
@Table(
	name = "chat_friendships",
	indexes = [
		Index(name = "idx_user_id", columnList = "user_id"),
		Index(name = "idx_friend_id", columnList = "friend_id"),
		Index(name = "idx_user_status", columnList = "user_id, status")
	],
	uniqueConstraints = [
		UniqueConstraint(name = "uk_friendship", columnNames = ["user_id", "friend_id"])
	]
)
data class ChatFriendshipEntity(
	@Id
	@Column(name = "id", length = 36, nullable = false)
	val id: String,

	@Column(name = "user_id", length = 36, nullable = false)
	val userId: String,

	@Column(name = "friend_id", length = 36, nullable = false)
	val friendId: String,

	@Enumerated(EnumType.STRING)
	@Column(name = "status", length = 20, nullable = false)
	var status: FriendshipStatus,

	@Column(name = "nickname", length = 100)
	var nickname: String? = null,

	@Column(name = "favorite", nullable = false)
	var favorite: Boolean = false,

	@Column(name = "created_at", nullable = false)
	val createdAt: Instant,

	@Column(name = "updated_at", nullable = false)
	var updatedAt: Instant
)
